sdk/core/src/tasks/sboms/github/sync.rs

```diff
@@ -0,0 +1,409 @@
+    format!("pkg:harbor:{}@{}", full_name, commit_hash)
+}
+
+/// Synchronizes SBOMS for a GitHub Group with Harbor.
+#[derive(Debug)]
+pub struct SyncTask {
+    mongo_svc: GitHubProviderMongoService,
+    pub(in crate::tasks::sboms::github) github: GitHubService,
+    sboms: SbomService,
+}
+
+#[async_trait]
+impl TaskProvider for SyncTask {
+
+    /// Builds the Packages Dependencies, Purls, and Unsupported from the GitHub API.
+    async fn run(&self, task: &mut Task) -> Result<HashMap<String, String>, Error> {
+
+        let mut errors = HashMap::new();
+
+        let mut repos: Vec<Repo> = match self.github.get_repos().await {
+            Ok(value) => value,
+            Err(err) => {
+                println!("Error attempting to get the repos: {}", err);
+                task.err_total += 1;
+                Vec::new()
+            }
+        };
+
+        let mut total = 0;
+        let mut failed = 0;
+        let mut skipped = 0;
+
+        for repo in &mut repos {
+
+            let name = repo.full_name.clone().unwrap_or(
+                String::from("name/missing")
+            );
+
+            let url = match repo.html_url.clone() {
+                Some(url) => url,
+                None => continue
+            };
+
+            if !SyncTask::should_skip(repo, name, url.clone()) {
+
+                match self.process_repo(repo, task).await {
+
+                    Ok(option) => match option {
+                        Some(raw) => {
+                            match self.sboms.ingest(
+                                    raw.as_str(),
+                                    None,
+                                    SbomProviderKind::GitHub,
+                                    Xref {
+                                        kind: XrefKind::Product,
+                                        map: HashMap::new()
+                                    },
+                                    task,
+                                )
+                                .await {
+                                Ok(sbom) => {
+                                    let opt = Some(sbom);
+                                    println!("==> SBOM Ingestion went OK");
+                                    opt
+                                },
+                                Err(err) => {
+                                    task.err_total += 1;
+                                    errors.insert(url.clone(), err.to_string());
+                                    println!("SBOM Ingestion FAILED");
+                                    failed += 1;
+                                    None
+                                }
+                            };
+                        }
+                        None => println!("==> No work to do because the hashes matched")
+                    }
+                    Err(err) => println!("==> Repo processing failure: {}", err)
+                };
+            } else {
+                skipped += 1;
```

Comment:
Collaboration point:  Not a blocker at all, but I want to dig into why things get skipped in the next sprint and figure out if we need to track that similar to how we track `Unsupported` in Snyk.

Reply:
We probably do.  If you want to know all of the reasons GitHub provider skipps things, look at the function: should_skip() in the GitHub SyncTask